src/UI/MainWindow.vala

```diff
@@ -50,6 +50,11 @@ public class Spreadsheet.UI.MainWindow : ApplicationWindow {
                 scrolled.add (viewport);
 
                 var sheet = new Sheet (page);
+                foreach (var cell in page.cells) {
+                    if (cell.selected) {
+                        style_popup.add (new StyleModal (cell.font_style, cell.cell_style));
```

Comment:
you could probably add a `break;` after that. It will avoid to loop over all the cells of the sheet, while the selected one is (always?) the first.

Reply:
Fixed!